synfig-studio/src/gui/asyncrenderer.cpp

```diff
@@ -800,26 +801,48 @@ AsyncRenderer::stop()
 			//Glib::MainContext::get_default()->iteration(false);
 
 			std::string error_message;
-			if(status == RENDERING_SUCCESS)
-				signal_success_();
-			else
-				error_message = _("Animation couldn't be rendered");
-
-			target=0;
-			render_thread=0;
 
-			lock.release();
+			if(interaction == UNDEFINED)
+			{
+				if(status == RENDERING_SUCCESS)
+					signal_success_();
+				else
+					error_message = _("Animation couldn't be rendered");
+
+				target=0;
```

Comment:
Honestly, not sure what those three lines do...

Reply:
`error_message = _("Animation couldn't be rendered");` ?
It "prepends" the error message that would be filled via line 819
`				error_message += "\n" + logger->get_error_message();`

`target = 0`is used to free the memory of `target` (it should be nullptr instead of 0 or NULL since C++11).

Reply:
This part here:
https://github.com/synfig/synfig/blob/d750262352a6bfb66b5fcc9ec092d008c41249f7/synfig-studio/src/gui/asyncrenderer.cpp#L812-L814

So I didn't touch it as for now.

Reply:
Got it.
I think only the 
`if(status == RENDERING_ERROR) { ...} ` below should be guarded by `if (interaction ....)`.

No matter why render should be stopped, there should be no target and no render_thread anymore (and lock must be releases). `target` means the exporter/encoder.